src/Workspaces/CSharpTest/Formatting/FormattingTests.cs

```diff
@@ -9498,5 +9498,64 @@ enum TestEnum
 }", changedOptionSet: changingOptions);
         }
 
+        [Fact, Trait(Traits.Feature, Traits.Features.Formatting)]
+        [WorkItem(36913, "https://github.com/dotnet/roslyn/issues/36913")]
+        public async Task NewLinesForBraces_SwitchExpression_Default()
+        {
+            await AssertFormatAsync(
+                @"
+class A
+{
+    void br()
+    {
+        var msg = 1 switch
+        {
+            _ => null
+        };
+    }
+}",
+                @"
+class A
+{
+    void br()
+    {
+        var msg = 1 switch {
+            _ => null
+        };
+    }
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.Formatting)]
+        [WorkItem(36913, "https://github.com/dotnet/roslyn/issues/36913")]
+        public async Task NewLinesForBraces_SwitchExpression_NonDefault()
+        {
+            var changingOptions = new Dictionary<OptionKey, object>
+            {
+                { NewLinesForBracesInControlBlocks, false },
+            };
+            await AssertFormatAsync(
+                @"
+class A
+{
+    void br()
+    {
+        var msg = 1 switch {
+            _ => null
+        };
+    }
+}",
+                @"
+class A
+{
+    void br()
+    {
+        var msg = 1 switch
+{
```

Comment:
❔ Is there a reason this is not indented?

Reply:
TBH I started out in the test `NewLineForOpenBracesNonDefault()`, where most of input control blocks aren't indented. However, during development, a separate test case worked much better for me, when I was debugging how everything worked in the formatting engine.

I thought it makes most sense to keep a separate test like this, but didn't consider oddness of the indentation. I think it makes most sense to change it to being indented. Shall I change it to that?

Reply:
It seems like this one brace should be indented, as that's the most likely scenario for formatting.

Reply:
```suggestion
        {
```